src/history/hs_rec.c

```diff
@@ -669,6 +669,26 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
                 F_SET(first_non_ts_upd, WT_UPDATE_CLEARED_HS);
             }
 
+            /* We should never write a prepared update to the history store. */
+            WT_ASSERT(session,
+              upd->prepare_state != WT_PREPARE_INPROGRESS &&
+                upd->prepare_state != WT_PREPARE_LOCKED);
+
+            /*
+             * Skip the check if we don't have a snapshot. Therefore, the check doesn't ensure that
+             * we never write uncommitted updates to the history store. But this is the best we can
+             * do for now.
+             *
+             * Sometimes the application threads and the checkpoint thread will fall behind the
+             * eviction threads, it may choose an invisible update to write to the disk if a failed
+             * eviction decided to write that update to the disk before. In this case, no point to
+             * check the visibility of the history store updates.
+             */
```

Comment:
Can you reword it to something like this:

Sometimes the application and the checkpoint threads will fall behind the eviction threads, and they will choose an invisible update that was previously selected by an eviction pass. In that case, there is no point in checking the visibility of such history store updates.

Reply:
reworded